### PR TITLE
[layer] Add script for prepare whiteouts

### PR DESCRIPTION
**Pull request by MykolaSolyanko**
Signed-off-by: Mykola Solianko <mykola_solianko@epam.com>

### CODE REVIEW

File: classes/layer-generator.bbclass [hunk 1]
```diff
@@ -59,6 +59,35 @@ do_create_rootfs_archive() {
...
 }
 
+python do_create_whiteouts() {
+    import os
+
+    whiteoutPrefix = ".wh."
+    whiteoutOpaqueDir = whiteoutPrefix + whiteoutPrefix + ".opq"
+
+    whiteouts = d.getVar("LAYER_WHITEOUTS_{}".format(d.getVar("PN")))
+    if whiteouts is None:
```
Comment (al1img) [inline classes/layer-generator.bbclass:69]: Do you need this check? The following loop will not be executed if whiteouts is None. Right?

File: classes/layer-generator.bbclass [hunk 2]
```diff
@@ -59,6 +59,35 @@ do_create_rootfs_archive() {
     mv ${IMAGE_ROOTFS_TAR} ${LAYER_WORK_DIR}/${DIGEST}
 }
 
+python do_create_whiteouts() {
+    import os
+
+    whiteoutPrefix = ".wh."
+    whiteoutOpaqueDir = whiteoutPrefix + whiteoutPrefix + ".opq"
```
Comment (al1img) [inline classes/layer-generator.bbclass:66]: You mask whiteoutOpaqueDir by whiteoutPrefix. If only whiteoutPrefix has to be changed then whiteoutOpaqueDir will be changed implicitly.

File: classes/layer-generator.bbclass (hunk 1 above)
Comment (MykolaSolyanko) [reply classes/layer-generator.bbclass:69]: If we do not add a check in this case, then when we try to split a string,  we will get an error that such an operation cannot be performed for TypeNone. If we don't add this check, then in each layer we will need to write this variable if we don't use whiteouts at all to reduce layer compiler error

Comment (al1img) [reply classes/layer-generator.bbclass:69]: ok.

File: classes/layer-generator.bbclass (hunk 2 above)
Comment (MykolaSolyanko) [reply classes/layer-generator.bbclass]: Done

**Pull request merged**